typescript/schedules.md

```diff
@@ -0,0 +1,1246 @@
+  | 'FRI'
+  | 'FRIDAY'
+  | 'SA'
+  | 'SAT'
+  | 'SATURDAY';
+
+export type DayOfWeekValue = ZeroTo7 | DayOfWeekName;
+
+// TODO can we do something to make it case insensitive?
+// export type DayOfWeekValue<T> = ZeroTo7 | Uppercase<T> extends DayOfWeekName ? T : DayOfWeekName
+export type DayOfWeek = Range<DayOfWeekValue> | DayOfWeekValue;
+
+/**
+ * An event specification relative to the calendar, similar to a traditional cron specification.
+ *
+ * A second in time matches if all fields match. This includes `dayOfMonth` and `dayOfWeek`.
+ */
+export interface CalendarSpec {
+  /**
+   * @default 0
+   */
+  second?: Second | Second[];
+
+  /**
+   * @default 0
+   */
+  minute?: Minute | Minute[];
+
+  /**
+   * @default 0
+   */
+  hour?: Hour | Hour[];
+
+  /**
+   * @default {@link AnyValue}
+   */
+  dayOfMonth?: DayOfMonth | DayOfMonth[];
+
+  /**
+   * @default {@link AnyValue}
+   */
+  month?: Month | Month[];
+
+  /**
+   * @default {@link AnyValue}
+   */
+  year?: Year | Year[];
+
+  /**
+   * Can accept 0 or 7 as Sunday.
+   *
+   * @default {@link AnyValue}
+   */
+  dayOfWeek?: DayOfWeek | DayOfWeek[];
+}
+
+/**
+ * IntervalSpec matches times that can be expressed as:
+ *
+ * `Epoch + (n * every) + at`
+ *
+ * where `n` is any integer ≥ 0.
+ *
+ * For example, an `every` of 1 hour with `at` of zero would match every hour, on the hour. The same `every` but an `at`
+ * of 19 minutes would match every `xx:19:00`. An `every` of 28 days with `at` zero would match `2022-02-17T00:00:00Z`
+ * (among other times). The same `every` with `at` of 3 days, 5 hours, and 23 minutes would match `2022-02-20T05:23:00Z`
+ * instead.
+ */
+export interface IntervalSpec {
+  every: Seconds;
+
+  /**
+   * @default 0
+   */
+  at?: Seconds;
+}
+
+/**
+ * A complete description of a set of absolute timestamps (possibly infinite) that an Action should occur at. These times
+ * never change, except that the definition of a time zone can change over time (most commonly, when daylight saving
+ * time policy changes for an area). To create a totally self-contained `ScheduleSpec`, use UTC.
+ */
+export interface ScheduleSpec {
+  /** Calendar-based specifications of times. */
+  calendars?: CalendarSpec[];
+
+  /** Interval-based specifications of times. */
+  intervals?: IntervalSpec[];
+
+  /** Any timestamps matching any of the exclude_calendar specs will be skipped. */
+  skip?: CalendarSpec[];
+
+  /**
+   * Any timestamps before `startAt` will be skipped. Together, `startAt` and `endAt` make an inclusive interval.
+   *
+   * @default The beginning of time
+   */
+  startAt?: Date;
+
+  /**
+   * Any timestamps after `endAt` will be skipped.
+   *
+   * @default The end of time
+   */
+  endAt?: Date;
+
+  /**
+   * All timestamps will be incremented by a random value from 0 to this amount of jitter.
+   *
+   * @default 1 second
+   */
+  jitter?: Milliseconds;
+
+  // Add to SDK if requested by users:
+  //
+  // string timezone_name = 10;
```

Comment:
timezone_name will definitely be requested. timezone_data not so much.

Reply:
Adding:

```ts
  /**
   * IANA timezone name, for example `US/Pacific`.
   * 
   * https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
   * 
   * @default `UTC`
   */
  timezone?: string;
```